File: python/LinkedList/collection/utils/filemanager.py
from ..ds.linkedlist import LinkedList

class FileManager :
    
    # Save linked list to a file
    def serialize(self, filename,list):
        current= list.head 
        with open(filename, "w") as file :
            while current != None:
                file.write(str(current.data)+ "->")
                current = current.next 

    # Load linked list from a file
    def deserialize(self, filename):
        list = LinkedList()
        try:
            with open(filename, "r") as file :
                data = file.read()
                items = data.split("->")
                for item in items:
                    if item :
                        list.insert(item)
        except FileNotFoundError:
            pass
        return list

File: C/LinkedList/src/filemanager.c
#include "../include/filemanager.h"
void serialize(const char* filename, struct LinkedList *list){
    FILE *file = fopen(filename, "w");
    struct Node *ptrcurrent = list->head ;
    while (ptrcurrent != NULL)
    {
        fprintf(file , "%d -> ", ptrcurrent->item);
        ptrcurrent = ptrcurrent->ptrnext ;
    }
    fclose(file);
}

struct LinkedList* deserialize(const char* filename){
    FILE *file =fopen(filename , "r");
    struct LinkedList *list = malloc(sizeof(struct LinkedList));
    initList(list);
    int item;
    while (fscanf(file , "%d -> " , &item) != EOF)
    {
        insert(list, item);
    }
    fclose(file);
    return list;
}